prog_test.go

```diff
@@ -360,6 +360,41 @@ func TestProgramGetNextID(t *testing.T) {
 	}
 }
 
+func TestProgramGetFDByID(t *testing.T) {
+	testutils.SkipOnOldKernel(t, "4.13", "bpf_prog_get_fd_by_id")
+
+	prog, err := NewProgram(&ProgramSpec{
+		Type: SkSKB,
+		Instructions: asm.Instructions{
+			asm.LoadImm(asm.R0, 0, asm.DWord),
+			asm.Return(),
+		},
+		License: "MIT",
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer prog.Close()
+	var next ProgramID
+
+	if next, err = ProgramGetNextID(ProgramID(0)); err != nil {
+		t.Fatal("Can't get next ID:", err)
+	}
+	if next == ProgramID(0) {
+		t.Fatal("Expected next ID other than 0")
+	}
+
+	if _, err = NewProgramFromID(next); err != nil {
```

Comment:
How much work is it to add `Program.ID() (ProgramID, error)` and similar for `Map`? Then this test could be a lot simpler:
```
id, err := prog.ID()
NewProgramFromID(id)
```

Reply:
I've added the implementation for `ID()` and updated the tests.